cpp/src/slice2js/JsUtil.cpp

```diff
@@ -164,6 +164,24 @@ fixIds(const StringList& ids)
     return newIds;
 }
 
+string
+Slice::JsGenerator::getDefinedIn(const ContainedPtr& p)
+{
+    static const string prefix = "js:defined-in:";
+    string definedIn;
+    if(findMetaData(prefix, p->getMetaData(), definedIn))
+    {
+        string base = p->definitionContext()->filename();
+        string::size_type pos = base.find_last_of("/\\");
+        if(pos != string::npos)
```

Comment:
Should be add an assert if no / or \ is found? In this case, the resulting string would not formed correctly.

Reply:
I think we can add the assert, the filename is always a full-path if I'm not mistaken, I will check.